### DISCUSSION TITLE
Sequencing in F' FSW

### DISCUSSION THREAD

**Discussion by zlynch2** [nasa/fprime#2982] [Q&A] [answered]
Hello F' Community, 

I have created an F' project and have successfully been able to send my component noop cmds through f' GDS. I know Fprime-gds has a notion of sequencing however I am having trouble following the sequencing example. Specifically, when I build fprime using fprime-util build I do not build the "fprime/Ref/build-artifacts/*/dict/RefTopologyAppDictionary.xml". I believe I am on v3.4.3 of F'.

Also, I was wondering if I can utilize command sequencing directly in F' fsw? I would like to send a command from my F' component every couple seconds and have the command sequence handler do that without having to go through F' GDS. 

If there is an example or place in documentation I have overlooked please let me know. I appreciate any feedback. Thank you!

**Comment by LeStarch** [comment]
What sequencing example are you referencing? 

`fprime/Ref/build-artifacts/*/dict/RefTopologyAppDictionary.xml` should likely read `fprime/Ref/build-artifacts/*/dict/<insert-your-deployment-name>TopologyAppDictionary.xml`.

It looks from the CmdSequencer.fpp, you can run sequence from other components by using the following port:

```
    @ Port for requests to run sequences
    async input port seqRunIn: Svc.CmdSeqIn
```

You'd need the output match to the above port.

**Reply by zlynch2** [reply]
I am refereeing to the "compiling a sample sequence" section in the F' docs located here: https://nasa.github.io/fprime/v3.3.0/UsersGuide/gds/seqgen.html#time-details

I am not sure how to create build that topologyAppDictionary.xml file before running my fprime-seqgen command.

**Reply by LeStarch** [reply, marked as answer]
When you run `fprime-util build` in your deployment, you should get a folder called "build-artifacts" in the root of your project.  Inside that folder there will be a tree like:

```
build-artifacts/<platform/<Deployment Name>/dict/<Deployment Name>TopologyAppDictionary.xml
```

For a deployment called "MyApp" built for Linux:
```
build-artifacts/Linux/MyApp/dict/MyAppTopologyAppDictionary.xml
```
